staging/src/k8s.io/kube-aggregator/pkg/registry/apiservice/strategy.go

```diff
@@ -25,19 +25,37 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/apiserver/pkg/registry/generic"
+	"k8s.io/apiserver/pkg/registry/rest"
 	"k8s.io/apiserver/pkg/storage"
 	"k8s.io/apiserver/pkg/storage/names"
 
 	"k8s.io/kube-aggregator/pkg/apis/apiregistration"
 	"k8s.io/kube-aggregator/pkg/apis/apiregistration/validation"
 )
 
+// RESTCreateUpdateStrategy is the base interface for our API service strategies.
+type RESTCreateUpdateStrategy interface {
```

Comment:
we have interfaces for this already, compare https://github.com/kubernetes/kubernetes/blob/d1031b18fc83146dd8418c17df289d09d10d29e5/staging/src/k8s.io/apiserver/pkg/registry/rest/update.go#L37

Reply:
@sttts Good catch!
What I'm trying to do here is to have a union of 2 interfaces: `rest.RESTUpdateStrategy` and `rest.RESTCreateStrategy`. However, these 2 interfaces have some methods in common so I can't embed the two of them into a single interface.

What I've just done now is to embed the `rest.RESTUpdateStrategy` interface and copy 2 other methods from the `rest.RESTCreateStrategy` to create a new `RESTCreateUpdateStrategy`

Reply:
I remember. Golang....

Can we move that to k8s.io/apiserver/pkg/registry/rest?